src/pf-touchspin/pf-touchspin.component.js

```diff
@@ -0,0 +1,359 @@
+    });
+
+    input.addEventListener('keyup', function (event) {
+      let keycode = event.keyCode ? event.keyCode : event.which;
+      if (keycode === 38) {
+        self._stop();
+      } else if (keycode === 40) {
+        self._stop();
+      }
+    });
+
+    // support for click foe down spin
+    down.addEventListener('mousedown', function (event) {
+      if (input.classList.contains('disabled')) {
+        return;
+      }
+      self._down();
+      self._downSpin();
+
+      event.preventDefault();
+      event.stopPropagation();
+    });
+
+    document.addEventListener('mouseup', function (event) {
+
+      event.preventDefault();
+      self._stop();
+    });
+
+    up.addEventListener('mousedown', function (event) {
+      if (input.classList.contains('disabled')) {
+        return;
+      }
+
+      self._up();
+      self._upSpin();
+
+      event.preventDefault();
+      event.stopPropagation();
+    });
+
+    // stop spinning if mouse is not over buttons
+    down.addEventListener('mouseout', function (event) {
+      event.stopPropagation();
+      self._stop();
+    });
+
+    up.addEventListener('mouseout', function (event) {
+
+      event.stopPropagation();
+      self._stop();
+    });
+
+    //support for mouse scroll
+    document.addEventListener('wheel', function (event) {
+      var delta = -event.deltaY;
+      if (input !== document.activeElement) {
+        return;
+      }
+      event.stopPropagation();
+      event.preventDefault();
+      if (delta < 0) {
+        self._down();
+      } else {
+        self._up();
+      }
+    });
+  }
+
+  /**
+  * Called when element's attribute value has changed
+  *
+  * @param {string} attrName The attribute name that has changed
+  * @param {string} oldValue The old attribute value
+  * @param {string} newValue The new attribute value
+  */
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    this.init();
+  }
+
+  /**
+   * Called when an instance of the element is created
+   */
+  constructor() {
+    super();
+  }
+
+  /**
+   * force the valur to be divisible by step
+   *
+   * @param {number} value
+   */
+  _stepDivisibility(value) {
+    switch (this._forceStepDivisibility) {
+      case 'round':
+        return (Math.round(value / this._step) * this._step).toFixed(this._decimals);
+      case 'floor':
+        return (Math.floor(value / this._step) * this._step).toFixed(this._decimals);
+      case 'ceil':
+        return (Math.ceil(value / this._step) * this._step).toFixed(this._decimals);
+      default:
+        return value;
+    }
+  }
+
+  /**
+   * check the value before change in value
+   */
+  _checkValue() {
+    var val, parsedval, returnval;
+
+    val = this.querySelector('input').value;
+
+    if (this._decimals > 0 && val === '.') {
+      return;
+    }
+
+    parsedval = parseFloat(val);
+
+    if (isNaN(parsedval)) {
+      parsedval = 0;
+    }
+
+    returnval = parsedval;
+
+    if (parsedval.toString() !== val) {
```

Comment:
if this conditional check necessary? we are setting `returnval = parseval` in statement above regardless, so i think maybe we can remove this if clause?